### PR TITLE
[hw,keymgr_dpe] Sync keymgr_dpe to master

### PR DESCRIPTION
**Pull request by Razer6**
This PR pulls over the Keymgr DPW from the integrated_dev repository. It only includes changes from `hw/ip/keymgr_dpe` modulo some documentation changes to include the DMA into the list of supported repositories and adding the `REGWEN_MUBI` as a countermeasure to the infrastructure.

This PR was created with:
```
git format-patch -k --stdout integrated_dev...master -- hw/ip/keymgr_dpe | git am -3 -k
```
and a little bit of manual rewriting of the history for the aforementioned changes.

### CODE REVIEW

**Comment by martin-velay** [comment]
Hi, I've used a similar format as Rupert [here](https://github.com/lowRISC/opentitan/pull/24434#issuecomment-2368101150) but I used a CLI tool from Andreas (thanks to both of you!).

The following command is giving the results I will comment below:
```
  ./mig_pr_review.py --source-branch integrated_dev --target-branch master --migration-branch Razer6/keymgr-sync-to-master --path hw/ip/keymgr_dpe
```

# List of commits which have been added compare to the `integrated_dev` branch
```
  WARNING: Fewer original commits (90) than migration commits (95); the following commits may have been added:
      eaf21ea745ea [keymgr,keymgr_dpe] Don't block PRNG reseeding operations while processing commands
      29851a86467c [keymgr,keymgr_dpe] Change reseed counter to count PRNG updates instead of cycles
      f3030f537cec [keymgr] Don't update and reseed PRNG in Disabled/Invalid state forever
      a3e9d978a87b [hw,keymgr_dpe] Update source files with new license header
      283d048ee523 [hw,keymgr_dpe] Update bazel rules for generating C/Rust headers
```

## [keymgr,keymgr_dpe] Don't block PRNG reseeding operations while processing commands ([eaf21ea](https://github.com/lowRISC/opentitan/pull/24438/commits/eaf21ea745ead1aa0733dc294d4d39674ac0095d))
RTL impact --> Previously, the `seed_en` signal was transmitted to the input port of `prim_lfsr` only when `~reg2hw.start.q`. This condition has been removed as no reason was found to keep it, and it seems to be more efficient from an entropy consumption point of view.
**--> OK**

## [keymgr,keymgr_dpe] Change reseed counter to count PRNG updates instead of cycles ([29851a8](https://github.com/lowRISC/opentitan/pull/24438/commits/29851a86467c31d25860f2a067d353df3417b3b7))
RTL impact --> Update the reseed counter only when the PRNG is updated.
This change has been done on the `master` and has some impacts on the `integrated_dev` branch for the new piece of code. Naturally, it also has an impact on DV.
**--> OK**

## [keymgr] Don't update and reseed PRNG in Disabled/Invalid state forever ([f3030f5](https://github.com/lowRISC/opentitan/pull/24438/commits/f3030f537cec19ca610183d27ce4ea486bf202a9))
RTL impact --> Limit the number PRNG updates until two more PRNG reseed operation have happened when we are in StCtrlDisabled or StCtrlInvalid states.
**--> OK**

## [hw,keymgr_dpe] Update source files with new license header ([a3e9d97](https://github.com/lowRISC/opentitan/pull/24438/commits/a3e9d978a87b9e34c672e22fe5ffd4a0a0db6505))
Only impacts on the comments.
**--> OK**

## [hw,keymgr_dpe] Update bazel rules for generating C/Rust headers ([283d048](https://github.com/lowRISC/opentitan/pull/24438/commits/283d048ee523efcdb3480330cf392120d955dd6b))
No RTL impact.
**--> OK**


# List of commits which differ between the `integrated_dev` branch and this PR
```
  ------------------------             integrated_dev               -----------------------------             this PR             ---------
  WARNING: Original commit b7d67c6b7fd8af3c3b81dd25aafda85eda42e6a8 differs from migration commit 5da1d47cca40a8469a8a15cc29a25b590cb1f9fe.
  WARNING: Original commit 59307424b6a1a30cc6338a934f93d96e2a681676 differs from migration commit 33ee5c32652651225c072d590b44b3aeb00e2ae6.
  WARNING: Original commit df23a95bff8f14e573cad46175e2ae5cfcae8320 differs from migration commit 37d369106044fba0d7eba42af6256ed18957064e.
  WARNING: Original commit 5ef0aef112d6819a3fbce2b058186492fc000f3b differs from migration commit 1f124212766472abc95d73cab114b125b1dce3fe.
  WARNING: Original commit a04f160c1b7cdd556ec898c0ab0f14569e098254 differs from migration commit 1a08e4f38fa8ce3bdb27f1771eec9da6f60466bc.
  WARNING: Original commit 4eec71275a0eb683545c9d3ebfc3fb685e8232db differs from migration commit a68d5a5d01ab210dd8e3481138cd769a458b1fb4.
```

## [keymgr_dpe, spec] Bring initial spec files ([5da1d47](https://github.com/lowRISC/opentitan/pull/24438/commits/5da1d47cca40a8469a8a15cc29a25b590cb1f9fe))
- `opentitan/SUMMARY.md`: I don't see the "Key Manager DPE" section as for the other IPs, even in the latest commit.
**--> action required (not blocking as it concerns only doc change)**

## [keymgr_dpe] Copy RTL files from keymgr to keymgr_dpe IP folder ([33ee5c3](https://github.com/lowRISC/opentitan/pull/24438/commits/33ee5c32652651225c072d590b44b3aeb00e2ae6))
- `opentitan/hw/ip/keymgr_dpe/README.md`: this file has been taken over but it's empty, it contains only the title.
**--> action required (not blocking as it concerns only doc change)**

## [keymgr_dpe] Autogenerate reg files with Makefile ([37d3691](https://github.com/lowRISC/opentitan/pull/24438/commits/37d369106044fba0d7eba42af6256ed18957064e))
- `opentitan/hw/Makefile`: keymgr_dpe is missing in this commit, but it has been taken over via the commit [#37d3691](https://github.com/lowRISC/opentitan/pull/24438/commits/37d369106044fba0d7eba42af6256ed18957064e)
**--> OK**

## [keymgr_dpe] Assign cip_id to 39 for keymgr_dpe ([1f12421](https://github.com/lowRISC/opentitan/pull/24438/commits/1f124212766472abc95d73cab114b125b1dce3fe))
- `opentitan/util/ip_block.py`: this file is not existing on the `integrated_dev` branch but the `master` branch already contains the reference to this block `keymgr_dpe` since this commit [#16bc8ec](https://github.com/lowRISC/opentitan/commit/16bc8ecfaeca634081044eb43ea8975fd2901768)
**--> OK**

## [keymgr_dpe] Enable RTL lint setup ([1a08e4f](https://github.com/lowRISC/opentitan/pull/24438/commits/1a08e4f38fa8ce3bdb27f1771eec9da6f60466bc))
- `opentitan/hw/top_darjeeling`: on the `integrated_dev` branch it seems that this [directory](https://github.com/lowRISC/opentitan/tree/integrated_dev/hw/top_darjeeling) contains many different files but this PR doesn't seems to take anything over into this [directory(https://github.com/lowRISC/opentitan/tree/5da1d47cca40a8469a8a15cc29a25b590cb1f9fe/hw/top_darjeeling)]. Is it a hole?
**--> action required**

## [rtl, prim] Add 'commit' functionality to prim_count ([a68d5a5](https://github.com/lowRISC/opentitan/pull/24438/commits/a68d5a5d01ab210dd8e3481138cd769a458b1fb4))
- This PR is mainly adding `commit_i` and `cnt_after_commit_o` ports which are on the `master` branch but were not present on the `integrated_dev`
**--> OK**

# Action items
* Update `opentitan/SUMMARY.md` file to bring over what is already on `integrated_dev` regarding Key Manager DPE
* Write a proper documentation here `opentitan/hw/ip/keymgr_dpe/README.md`
* Probably need to populate `opentitan/hw/top_darjeeling` directory with some relevant files as it is on the `integrated_dev` branch
* CI is failing, I guess these issues should be resolved before merging

**Review by martin-velay — CHANGES_REQUESTED** [review]
See this comment: https://github.com/lowRISC/opentitan/pull/24438#issuecomment-2371286449

**Comment by Razer6** [comment]
> * Update `opentitan/SUMMARY.md` file to bring over what is already on `integrated_dev` regarding Key Manager DPE

Added entries to `SUMMARY.md`

Regarding linting:

A PR for that is open in https://github.com/lowRISC/opentitan/pull/24625

**Comment by Razer6** [comment]
> * Write a proper documentation here `opentitan/hw/ip/keymgr_dpe/README.md`

I would defer that to an issue and not being a blocker for this PR, since it is not yet done on the integrated_dev branch as well.

**Comment by martin-velay** [comment]
> > * Write a proper documentation here `opentitan/hw/ip/keymgr_dpe/README.md`
> 
> I would defer that to an issue and not being a blocker for this PR, since it is not yet done on the integrated_dev branch as well.

Sure, I have already mentioned this as non-blocking in my comment (maybe the word "required" was not appropriate though:
```
opentitan/hw/ip/keymgr_dpe/README.md: this file has been taken over but it's empty, it contains only the title.
--> action required (not blocking as it concerns only doc change)
```

**Comment by Razer6** [comment]
@rswarbrick CI is green and the PR has been reviewed. Can you take a look getting it merged?

**Comment by rswarbrick** [comment]
Ah, cool. Thanks, @martin-velay for reviewing this. I'm approving the PR now but we'll need one extra committer to explicitly approve the change to some RTL (which is currently caught by the block list). I'll chase someone for that.

CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_ctrl.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_op_state_ctrl.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_pkg.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_reg_pkg.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_reg_top.sv

**Comment by vogelpi** [comment]
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_ctrl.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_op_state_ctrl.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_pkg.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_reg_pkg.sv
CHANGE AUTHORIZED: hw/ip/keymgr_dpe/rtl/keymgr_dpe_reg_top.sv

This PR touches RTL files not included in Earlgrey-PROD. This is fine.

**Pull request merged**
